worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/MessageDistributor.java

```diff
@@ -17,51 +17,159 @@
-            if(stagingQueues.isEmpty()) {
+
+            if (stagingQueues.isEmpty()) {
                 continue;
             }
-            
+
             LOGGER.debug("Creating a new DistributorWorkItem for target queue: {} and staging queues: {}", targetQueue, stagingQueues);
-            
+
             distributorWorkItems.add(new DistributorWorkItem(targetQueue, stagingQueues));
-            
+
         }
         return distributorWorkItems;
     }
+
+    private Set<Queue> getTargetQueues(final Connection connection, final List<Queue> queues)
+    {
+        // The set of target queues that already exist
+        final Set<Queue> targetQueuesAlreadyCreated = queues
+                .stream()
+                .filter(q -> !q.getName().contains(LOAD_BALANCED_INDICATOR))
+                .collect(Collectors.toSet());
+
+        // The set of target queues that need to be created (in the scenario where a worker is deployed initially with 0 instances its
+        // target queue will not be created, so when this happens we need to create the target queue, otherwise the messages will never
+        // get moved on from the staging queues)
+        final Set<String> targetQueuesToBeCreated = getTargetQueuesToBeCreated(queues, targetQueuesAlreadyCreated);
+
+        if (!targetQueuesToBeCreated.isEmpty()) {
+            LOGGER.info("Target queue(s) to be created: {}", targetQueuesToBeCreated);
+
+            // Create RabbitMQ channel
+            final Channel channel;
+            try {
+                channel = connection.createChannel();
+            } catch (final IOException e) {
+                throw new RuntimeException("IO Exception encountered trying to create a RabbitMQ channel", e);
+            }
+
+            // Create target queues
+            for (final String targetQueueToBeCreated : targetQueuesToBeCreated) {
+                try {
+                    LOGGER.info(String.format("Creating target queue by calling channel.queueDeclare(%s, %s, %s, %s, %s)",
+                                    targetQueueToBeCreated, targetQueueDurable, targetQueueExclusive, targetQueueAutoDelete,
+                            targetQueueArgs));
+
+                    channel.queueDeclare(
+                            targetQueueToBeCreated,
+                            targetQueueDurable,
+                            targetQueueExclusive,
+                            targetQueueAutoDelete,
+                            targetQueueArgs);
+                } catch (final IOException e) {
+                    throw new RuntimeException(String.format(
+                            "IOException thrown trying to create target queue when calling channel.queueDeclare(%s, %s, %s, %s, %s)",
+                            targetQueueToBeCreated, targetQueueDurable, targetQueueExclusive, targetQueueAutoDelete, targetQueueArgs), e);
+                }
+            }
+
+            // Close RabbitMQ channel
+            try {
+                channel.close();
```

Comment:
If `queueDeclare()` throws an `IOException` you won't attempt to close the channel.  This should be in a finally (or a try-with-resources) to make sure that it is always attempted.